commit 3068c4290166076093119d1dc08a2c0fada8235f
Author: Noah Watkins <nwatkins@redhat.com>
Date:   Thu Jul 26 10:52:34 2018 -0700

    mgr: report health check changes immediately
    
    Currently beacons and reports are sent periodically, which means that
    health check changes made by modules are only visible to the monitor
    after a delay. This patch sends a health report immediately after a
    change occurs.
    
    1. decouple scheduling of reports from beacons
    2. moves report scheduling into DaemonServer
    3. exposes a "send now" option
    4. health check change detection triggers immediate report
    
    Signed-off-by: Noah Watkins <nwatkins@redhat.com>

diff --git a/src/mgr/Mgr.cc b/src/mgr/Mgr.cc
index 40200605dc7..ca784ebcab7 100644
--- a/src/mgr/Mgr.cc
+++ b/src/mgr/Mgr.cc
@@ -286,7 +286,7 @@ void Mgr::init()
   // assume finisher already initialized in background_init
   dout(4) << "starting python modules..." << dendl;
   py_module_registry->active_start(daemon_state, cluster_state,
-      kv_store, *monc, clog, *objecter, *client, finisher);
+      kv_store, *monc, clog, *objecter, *client, finisher, server);
 
   dout(4) << "Complete." << dendl;
   initializing = false;
@@ -650,12 +650,6 @@ void Mgr::handle_mgr_digest(MMgrDigest* m)
   }
 }
 
-void Mgr::tick()
-{
-  dout(10) << dendl;
-  server.send_report();
-}
-
 std::map<std::string, std::string> Mgr::get_services() const
 {
   Mutex::Locker l(lock);
diff --git a/src/mgr/Mgr.h b/src/mgr/Mgr.h
index d684662a0c9..e5739c8c9a7 100644
--- a/src/mgr/Mgr.h
+++ b/src/mgr/Mgr.h
@@ -90,8 +90,6 @@ public:
 
   bool ms_dispatch(Message *m);
 
-  void tick();
-
   void background_init(Context *completion);
   void shutdown();
 
